src/Pagination/ConnectionField.php

```diff
@@ -35,17 +37,35 @@ public function pageInfoResolver(LengthAwarePaginator $paginator): array
      * Resolve edges for connection.
      *
      * @param  \Illuminate\Contracts\Pagination\LengthAwarePaginator  $paginator
+     * @param  array  $args
+     * @param  \Nuwave\Lighthouse\Support\Contracts\GraphQLContext  $context
+     * @param  GraphQL\Type\Definition\ResolveInfo  $resolveInfo
      * @return \Illuminate\Support\Collection
      */
-    public function edgeResolver(LengthAwarePaginator $paginator): Collection
+    public function edgeResolver(LengthAwarePaginator $paginator, $args, GraphQLContext $context, ResolveInfo $resolveInfo): Collection
     {
+        $typeFields = $resolveInfo->returnType->ofType->getFields();
         $firstItem = $paginator->firstItem();
 
-        return $paginator->values()->map(function ($item, $index) use ($firstItem): array {
-            return [
-                'cursor' => Cursor::encode($firstItem + $index),
-                'node' => $item,
-            ];
+        return $paginator->values()->map(function ($item, $index) use ($firstItem, $typeFields): array {
+            $data = [];
+
+            foreach ($typeFields as $field) {
+                switch ($field->name) {
+                    case 'cursor':
+                        $data['cursor'] = Cursor::encode($firstItem + $index);
+                        break;
+
+                    case 'node':
+                        $data['node'] = $item;
+                        break;
+
+                    default:
+                        $data[$field->name] = $item->pivot->{$field->name};
```

Comment:
Can paginators be used for resolving something else then models? `->pivot` might be undefined

Reply:
I was thinking of checking whether the field has a customer resolver directive. If it does, then it skips it, assuming they’ll handle it. Otherwise, the context (inside an edge) is the middle-space between entities, so I think the pivot table makes sense. Do you have any thoughts on how it should be handled when the dev adds a field that doesn’t have pivot data but doesn’t have a resolve directive, either?

Reply:
One simple option here would be to check for `$item->pivot` and silently move along if it doesn't exist, assuming they're handling the field resolving.

Reply:
Checking for `pivot` seems sufficient. Differentiating between what goes in the edge and what goes in the node might get complicated if we go for maximum flexibility, so let's see how far we get with that convention.